esp-hal/src/lcd_cam/mod.rs

```diff
@@ -31,22 +31,24 @@ pub struct LcdCam<'d, DM: crate::Mode> {
     pub lcd: Lcd<'d, DM>,
     /// The Camera interface.
     pub cam: Cam<'d>,
+
+    _guard: PeripheralGuard,
 }
 
 impl<'d> LcdCam<'d, Blocking> {
     /// Creates a new `LcdCam` instance.
     pub fn new(lcd_cam: impl Peripheral<P = LCD_CAM> + 'd) -> Self {
         crate::into_ref!(lcd_cam);
 
-        PeripheralClockControl::reset(system::Peripheral::LcdCam);
-        PeripheralClockControl::enable(system::Peripheral::LcdCam);
+        let guard = PeripheralGuard::new(system::Peripheral::LcdCam);
```

Comment:
Here you need to create two guards up front, one for cam and one for lcd.

(Might be worth having a version of `PeripheralGuard::new` that returns multiple guards at once, to avoid locking N times)

Reply:
> Here you need to create two guards up front, one for cam and one for lcd.

true 👍